pkg/promtail/targets/journaltargetmanager.go

```diff
@@ -0,0 +1,114 @@
+package targets
+
+import (
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/loki/pkg/logentry/stages"
+	"github.com/grafana/loki/pkg/promtail/api"
+	"github.com/grafana/loki/pkg/promtail/positions"
+	"github.com/grafana/loki/pkg/promtail/scrape"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// JournalTargetManager manages a series of JournalTargets.
+type JournalTargetManager struct {
+	targets map[string]*JournalTarget
+}
+
+// NewJournalTargetManager creates a new JournalTargetManager.
+func NewJournalTargetManager(
+	logger log.Logger,
+	positions *positions.Positions,
+	client api.EntryHandler,
+	scrapeConfigs []scrape.Config,
+) (*JournalTargetManager, error) {
+	tm := &JournalTargetManager{
+		targets: make(map[string]*JournalTarget),
+	}
+
+	for _, cfg := range scrapeConfigs {
+		if cfg.JournalConfig == nil {
+			continue
+		}
+
+		// TODO(rfratto): DRY the duplicate code from NewFileTargetManager?
+		registerer := prometheus.DefaultRegisterer
+		pipeline, err := stages.NewPipeline(log.With(logger, "component", "pipeline"), cfg.PipelineStages, &cfg.JobName, registerer)
+		if err != nil {
+			return nil, err
+		}
+
+		// Backwards compatibility with old EntryParser config
+		if pipeline.Size() == 0 {
```

Comment:
You can remove this entire block, the EntryParser config has been deprecated for long enough now, if someone specifies it I think it's fine to just be ignored by this targetmanager

Reply:
Done.